stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -497,7 +533,10 @@ function sparse(I::AbstractVector{Ti}, J::AbstractVector{Ti}, V::AbstractVector{
               "length(I) (=$(length(I))) == length(J) (= $(length(J))) == length(V) (= ",
               "$(length(V)))")))
     end
-
+    Tj = Ti
+    while isbitstype(Tj) && coolen >= typemax(Tj)
+        Tj = widen(Tj)
```

Comment:
This made `Tj` type unstable and caused the regression in https://github.com/JuliaLang/julia/issues/32985. Only matters for smallish arrays but would be good to fix nonetheless. Can't we just always use `Int` for this or something?